src/librustc_trans/trans/adt.rs

```diff
@@ -462,36 +503,38 @@ fn find_discr_field_candidate<'tcx>(tcx: &ty::ctxt<'tcx>,
             match field_ty.sty {
                 ty::TyRawPtr(ty::TypeAndMut { ty, .. }) if !type_is_sized(tcx, ty) => {
                     path.extend_from_slice(&[0, FAT_PTR_ADDR]);
-                    Some(path)
+                    Some((path, C_null(type_of::sizing_type_of(cx, ty))))
                 },
                 ty::TyRawPtr(..) | ty::TyInt(..) | ty::TyUint(..) => {
                     path.push(0);
-                    Some(path)
+                    Some((path, C_null(type_of::sizing_type_of(cx, ty))))
```

Comment:
Not sure I understand your sentence, but I just realized that `ty` was shadowed and that I should probably use the shadowed version instead. Right?